### PR TITLE
Style

### PR DESCRIPTION
**Pull request by zimbatm**
Just some suggestions. To be cherry-picked as pleased

/cc @jamescun

### CODE REVIEW

**Comment by zimbatm** [comment]
The last-one is probably the most interesting, the rest is just cosmetic

File: README.md [hunk 1]
```diff
@@ -82,14 +82,16 @@ usage: buddha [flags] job_file jobs...
...
   --version                   display version information
+  -y                          confirm run all
+  -h, --help                  shows this help
 
 examples:
   to invoke api_server from /etc/buddha.d:
     $ buddha api_server
   to invoke all jobs from /etc/buddha.d:
-    $ buddha all
+    $ buddha -y all
```
Comment (mdpye) [inline README.md]: Telling people to use `-y` here bypasses the warning that `-y` was introduced to deliver.

I'm not saying it was a great mechanism, but this makes it pointless unless it says "in alphabetical order, probably not what you want!" in large letters in the example description.

File: cmd/buddha.go [hunk 2]
```diff
@@ -16,40 +16,45 @@ var (
...
-	fmt.Print("  to invoke demo.json file:\r\n")
-	fmt.Print("    $ buddha --config=demo.json all\r\n")
-	fmt.Print("  to invoke jobs from stdin:\r\n")
-	fmt.Print("    $ cat demo.json | buddha --stdin all\r\n")
+	fmt.Fprint(os.Stderr, USAGE)
 }
 
 // --version
 func Version() {
-	fmt.Printf("Build Version: %s\r\n", BuildVersion)
```
Comment (mdpye) [inline cmd/buddha.go]: `\r`? Wha?!

**Comment by jamescun** [comment]
@zimbatm @mdpye I'm refactoring `main.go` as in its current state it is rather hard to test and the more recent changes (`-y`, running order etc) have made it a little messy.

I will incorporate these changes in to my branch.

File: README.md (hunk 1 above)
Comment (zimbatm) [inline README.md]: Until recently is was "in inode order" :p

As a user I just want to type `buddha` and have all my processes restarted if they are not on the latest code, isn't that the goal of the "all" option ? Each process would have a "before" that would abort if it's already running the latest version. Then I don't have to maintain the state of which machine and process is on what version and can run a simple `for machine in $(my_machines); do ssh $machine buddha -y all; done`, and restart the command if anything fails like my connectivity.

**Comment by zimbatm** [comment]
@jamescun I believe you still want to cherry-pick https://github.com/pusher/buddha/commit/4974d4fdc69d2bc2807ba63452019d7c6b2ad763 which fixes the tests for go 1.4

File: README.md (hunk 1 above)
Comment (mdpye) [inline README.md]: Agreed. Perhaps we should allow meta-jobs. @jamescun What about a job definition which just references a list of other jobs?

**Comment by jamescun** [comment]
Introduced in https://github.com/pusher/buddha/commit/db119198e94ed39359ca702a5fb3b17f8ac0f5b6

**Pull request closed without merging**